flux-driver/src/collector.rs

```diff
@@ -27,12 +27,19 @@ pub(crate) struct SpecCollector<'tcx, 'a> {
     error_guaranteed: Option<ErrorGuaranteed>,
 }
 
+/// An IgnoreKey is either `Some(module_def_id)` or `None` indicating the entire crate
+pub type IgnoreKey = Option<LocalDefId>;
```

Comment:
Maybe better to have an explicit enum

```suggestion
#[derive(PartialEq, Eq, Hash)]
pub enum IgnoreKey {
    Crate,
    Module(LocalDedIf)
}
```